qe/drivers.py

```diff
@@ -180,9 +181,42 @@ def gen_all_connected_determinant(self, psi_det: Psi_det) -> Psi_det:
                 if any(Excitation.is_connected(det_connected, d) for d in psi_det[:i]):
                     continue
 
-                l.append(det_connected)
+                l_global.append(det_connected)
+
+        # Return connected space
+        return l_global
+
+    def get_chunk_of_connected_determinants(self, psi_det: Psi_det):
+        """
+        Inputs:
+        :param psi_det: list of determinants
 
-        return l
+        MPI function, generates chunks of connected determinants
+
+        >>> d1 = Determinant((0, 1), (0,) ) ; d2 = Determinant((0, 2), (0,) )
+        >>> psi_chunk = Excitation(4).get_chunk_of_connected_determinants( [ d1,d2 ] )
+        >>> len(psi_chunk)
+        22
+        """
+        # Generate all connected determinants
+        psi_connected = self.gen_all_connected_determinant(psi_det)
+        world_size = MPI.COMM_WORLD.Get_size()
+        # TODO: len(psi_connected) will not scale, but is needed for this naive representation
+        floor, remainder = divmod(len(psi_connected), world_size)
+        ceiling = floor + 1
+        # Compute work distribution (size of chunk given to each node)
+        distribution = np.array(
+            [ceiling] * remainder + [floor] * (world_size - remainder), dtype="i"
+        )
+        # Compute offests (start of the local chunk)
+        offsets = np.zeros(world_size, dtype="i")
+        np.add.accumulate(distribution[:-1], out=offsets[1:])
+
+        return psi_connected[
+            offsets[MPI.COMM_WORLD.Get_rank()] : (
+                offsets[MPI.COMM_WORLD.Get_rank()] + distribution[MPI.COMM_WORLD.Get_rank()]
```

Comment:
may want to use `https://numpy.org/doc/stable/reference/generated/numpy.array_split.html#numpy.array_split` :)